gpiozero/output_devices.py

```diff
@@ -1185,24 +1186,37 @@ class Motor(SourceMixin, CompositeDevice):
         :class:`DigitalOutputDevice` instances, allowing only direction
         control.
 
+    :param bool enable_is_pwm:
+        If :data:`True`, construct a :class:`PWMOutputDevice` instance for
+        the enable pin, allowing variable speed control using the enable pin
+        and not the motor controller pins. Additionally, if this is
+        specified, *enable* must not be :data:`None` and *pwm* must be
+        :data:`False`. The default is :data:`False`.
+
     :type pin_factory: Factory or None
     :param pin_factory:
         See :doc:`api_pins` for more information (this is an advanced feature
         which most users can ignore).
     """
     def __init__(self, forward, backward, *, enable=None, pwm=True,
-                 pin_factory=None):
+                 enable_is_pwm=False, pin_factory=None):
+        if pwm and enable_is_pwm:
+            raise OutputDeviceError("Can't specify both PWM direction and enable pins")
+        if enable_is_pwm and enable is None:
+            raise OutputDeviceError("PWM enable requires a valid enable pin")
         PinClass = PWMOutputDevice if pwm else DigitalOutputDevice
         devices = OrderedDict((
             ('forward_device', PinClass(forward, pin_factory=pin_factory)),
             ('backward_device', PinClass(backward, pin_factory=pin_factory)),
         ))
-        if enable is not None:
+        if enable is not None and not enable_is_pwm:
             devices['enable_device'] = DigitalOutputDevice(
                 enable,
                 initial_value=True,
                 pin_factory=pin_factory
             )
+        elif enable is not None:
+            devices['enable_device'] = PWMOutputDevice(enable, pin_factory=pin_factory)
```

Comment:
Only a minor tweak, but IMHO this might look better without the duplicate `enable is not None` tests? i.e. something like:
```python3
        if enable is not None:
            if not enable_is_pwm:
                devices['enable_device'] = DigitalOutputDevice(
                    enable,
                    initial_value=True,
                    pin_factory=pin_factory
                )
            else:
                devices['enable_device'] = PWMOutputDevice(enable, pin_factory=pin_factory)
```